app/waf/src/controllers/adc.controller.ts

```diff
@@ -98,9 +104,9 @@ export class AdcController extends BaseController {
         );
       }
 
-      this.trustAdc(adc).then(() => {
+      await this.trustAdc(adc).then(async () => {
```

Comment:
May not need to wait here. POST request need to return, otherwise client may timeout.

Reply:
This session of code(HW) is just used to debug code, will eventually be removed. 

With await, it's easy to debug forwards. It's OK to remove it, no impact to me.

Reply:
Whatever removing or not, that is wrong implementation. We may not return response after install AS3 rpm. That takes about 30 seconds.

Reply:
When removing `await`, there's something wrong to acceptance test. 

```
    1) post /adcaas/v1/adcs: create ADC HW with trust timeout
(node:94655) UnhandledPromiseRejectionWarning: Error: Entity not found: Adc with id "76e59636-845e-4d12-b163-77e03af3bb13"
    at AdcRepository.findById (/Users/zong/GitRepos/zongzw/f5-adcaas-openstack/app/waf/src/repositories/common.ts:52:13)
    at processTicksAndRejections (internal/process/next_tick.js:81:5)
(node:94655) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 14)
(node:94655) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.

```

I give up on changing it back to the intermediate state.

Reply:
The current testing code in master brach supports this HW path. If your new change breaks the test, it will be your responsibility to to fix it. Or, if there is really no person need HW path to debug, we may delete this path, and re-enable HW path in the future.

Reply:
What does it mean "Whatever removing or not, that is wrong implementation. We may not return response after install AS3 rpm. ". Do you mean we should not await?

Reply:
> The current testing code in master brach supports this HW path. If your new change breaks the test, it will be your responsibility to to fix it. Or, if there is really no person need HW path to debug, we may delete this path, and re-enable HW path in the future.

I prefer to remove it, however, I'm not sure you or others still needs it. 
The related code is big(because it includes many test code). You can post a PR to remove it if necessary. 

The VE onboarding path's time cost is ~3 min. For the second/third/.. time is <1min. Should be convenient for testing already.

Reply:
I am fine to remove it. You may let others know that. Maybe Jikui or someone still use it.

Reply:
@jikui Maybe creating a standalone API for the developing is more agile than placing them in create application of 'HW'.

Reply:
> agile

Let's discuss this later when necessary.

Reply:
Let's make this HW path unchanged.

Reply:
Do you persist on this? 
I think we'll soon remove this `if`. -- HW. You also agree on this point. 

I don't think doing an investigation on it is valuable.

Reply:
We already talk that in the meeting... If you can make CI pass without modify this part of code, that should be better, because it will not impact others. I used to believe you can not make CI pass with your new refactoring change set, but you said you can....

Reply:
How about removing the HW path directly. I think no one is using it for now.

Reply:
It's no value to fix the code which is going away.